artpaint/layers/LayerView.cpp

```diff
@@ -28,76 +31,30 @@
-	:	BBox(BRect(0,0,120,LAYER_VIEW_HEIGHT-1),"a layer view",
-		B_FOLLOW_TOP|B_FOLLOW_LEFT,
+LayerView::LayerView(BBitmap *image, Layer *layer)
+	:	BBox("a layer view",
 		B_WILL_DRAW | B_NAVIGABLE_JUMP | B_FRAME_EVENTS, B_NO_BORDER)
 {
-	the_image = image;
 	the_layer = layer;
 
 	BMessage a_message;
-	a_message.AddInt32("layer_id",the_layer->Id());
-	a_message.AddPointer("layer_pointer",(void*)the_layer);
+	a_message.AddInt32("layer_id", the_layer->Id());
+	a_message.AddPointer("layer_pointer", (void*)the_layer);
 	a_message.what = HS_LAYER_VISIBILITY_CHANGED;
-	visibility_box = new BCheckBox(BRect(80,5,140,30),
-		"visibility check box",
-		B_TRANSLATE("Visible"),new BMessage(a_message));
-
-	BRect rect = visibility_box->Frame();
-	rect.OffsetBy(0,rect.Height()+4);
-
-//	The target should be set later, when we know that image-view is alrady in its window.
-	ResizeBy(visibility_box->Frame().right - 120+5,0);
-	AddChild(visibility_box);
-
-
-	layer_operation_pop_up_menu = new BPopUpMenu("layer operation menu");
-
-
-	a_message.what = HS_MERGE_WITH_UPPER_LAYER;
-	layer_operation_pop_up_menu->AddItem(new BMenuItem(
-		B_TRANSLATE("Merge with front layer"),
-		new BMessage(a_message)));
-
-	a_message.what = HS_MERGE_WITH_LOWER_LAYER;
-	layer_operation_pop_up_menu->AddItem(new BMenuItem(
-		B_TRANSLATE("Merge with back layer"),
-		new BMessage(a_message)));
+	visibility_box = new BCheckBox("visibility check box",
+		"", new BMessage(a_message));
 
-	layer_operation_pop_up_menu->AddSeparatorItem();
+	thumbnail_view = new ThumbnailView(image);
+	thumbnail_view->SetEventMask(0);
 
-	a_message.what = HS_ADD_LAYER_FRONT;
-	layer_operation_pop_up_menu->AddItem(new BMenuItem(
-		B_TRANSLATE("Add layer in front"),
-		new BMessage(a_message)));
+	layer_name_field = new BTextControl("", "Layer",
+		new BMessage(HS_LAYER_NAME_CHANGED));
 
-	a_message.what = HS_ADD_LAYER_BEHIND;
-	layer_operation_pop_up_menu->AddItem(new BMenuItem(
-		B_TRANSLATE("Add layer behind"),
-		new BMessage(a_message)));
-
-	layer_operation_pop_up_menu->AddSeparatorItem();
-
-	a_message.what = HS_DUPLICATE_LAYER;
-	layer_operation_pop_up_menu->AddItem(new BMenuItem(
-		B_TRANSLATE("Duplicate layer"),
-		new BMessage(a_message)));
-
-	layer_operation_pop_up_menu->AddSeparatorItem();
-
-	a_message.what = HS_DELETE_LAYER;
-	layer_operation_pop_up_menu->AddItem(new BMenuItem(
-		B_TRANSLATE("Delete layer"),
-		new BMessage(a_message)));
-
-	layer_operation_pop_up_menu->SetRadioMode(false);
-
-	popUpMenuField = new BMenuField(BRect(0,0,0,0),"popUpMenuField","",layer_operation_pop_up_menu);
-	popUpMenuField->ResizeToPreferred();
-	popUpMenuField->MoveTo(visibility_box->Frame().left,visibility_box->Frame().bottom+4);
-	popUpMenuField->MenuItem()->SetLabel("");
-	AddChild(popUpMenuField);
+	BGroupLayout* layerLayout = BLayoutBuilder::Group<>(this, B_HORIZONTAL, 5.0)
+		.Add(visibility_box)
+		.Add(thumbnail_view)
+		.Add(layer_name_field)
+		.SetInsets(5.0, 5.0, 5.0, 5.0);
```

Comment:
It may be a good idea to use constants from the InterfaceDefs.h for all insets, e.g. B_USE_SMALL_INSETS.